### PR TITLE
a few more typos for sip-004 and sip-005

### PR DESCRIPTION
**Pull request by psq**
Very interesting and useful read.  I learned a lot by reading in details, and will probably need to re-read many times...  Thank you for writing these SIPs, it helps in understanding the code further and how things are supposed to work.

I do have one question on https://github.com/psq/stacks-blockchain/blob/sip/sip-typos-2/sip/sip-005-blocks-and-transactions.md#calculating-the-state-of-an-account

instead of being just a `c32check-encoded address`, shouldn't this be either that, or a contract address, as contract can have balances as well.  Plus the implementation of `make_key_for_account` takes a `PrincipalData`, which supports both types.

### CODE REVIEW

**Comment by jcnelson** [comment]
> I do have one question on https://github.com/psq/stacks-blockchain/blob/sip/sip-typos-2/sip/sip-005-blocks-and-transactions.md#calculating-the-state-of-an-account
> instead of being just a c32check-encoded address, shouldn't this be either that, or a contract address, as contract can have balances as well. Plus the implementation of make_key_for_account takes a PrincipalData, which supports both types.

Since contract accounts can never send transactions, their nonce value is always zero, and that key/value pair gets materialized for contract accounts.  That's why the specification has only "c32check-encoded address" for nonces.  But you raise a good point for STX balances -- those keys can contain either standard principal addresses (just a c32check-encoded address) or contract principals.  This would also apply to fungible token balances and non-fungible token ownership -- either a standard principal or a contract principal can own these.  Want to add that minor correction to your PR?

**Comment by psq** [comment]
should be addressed in 4baba25.

File: sip/sip-005-blocks-and-transactions.md [hunk 1]
```diff
@@ -1229,13 +1229,15 @@ A non-fungible token owned by an account is encoded as follows:
...
 and the serialized Clarity value that represents the token.
-* Value: a serialized Clarity standard principal
+* Value: a serialized Clarity Principal (either a Standard Principal or a Contract Principal)
 
 Example: `"vm::SP13N5TE1FBBGRZD1FCM49QDGN32WAXM2E5F8WT2G.example-contract::4::example-nft::\x02\x00\x00\x00\x0b\x68\x65\x6c\x6c\x6f\x20\x77\x6f\x72\x6c\x64"` 
 refers to the non-fungible token `"hello world"` (which has type `buff` and is
 comprised of 11 bytes), defined in Clarity contract `SP13N5TE1FBBGRZD1FCM49QDGN32WAXM2E5F8WT2G.example-contract`
 as a type of non-fungible token `example-nft`.
 
+A Principal Address is either a 32check-encoded address in the case of standard principal, or a c32check-encoded address, followed by an ASCII period `.`0, and an ASCII-encoded string for a contract principal.
```
Comment (jcnelson) [inline sip/sip-005-blocks-and-transactions.md:1239]: Typo -- should be `c32check-encoded`.  Also, it looks like you have an extra 0 in "`.`0"

Comment (psq) [reply sip/sip-005-blocks-and-transactions.md]: thanks for catching this, embarrassing...  fixed in 30875ff

**Comment by jcnelson** [comment]
Thanks @psq!  These look good to me

**Pull request merged**